low-level/files/opt/seagate/sspl/setup/sspl_test.py

```diff
@@ -25,17 +33,47 @@ class SSPLTestCmd:
     def __init__(self, args: list):
         self.args = args
         self.name = "sspl_test"
+        self.plan = "self_primary"
+        self.avoid_rmq = False
 
     def process(self):
-        # TODO: Need to convert run_tests.sh from shell to python
-        test_plan=None
-        sspl_test_plans = ["sanity", "alerts", "self_primary","self_secondary"]
-        for i in range(len(self.args)):
-            if self.args[i] in sspl_test_plans:
-                test_plan = self.args[i]
-        if test_plan is None:
-            test_plan="self_primary"
-        CMD = f"{TEST_DIR}/run_tests.sh test {test_plan}"
+        global_config = self.args.config[0]
+        self.plan = self.args.plan[0]
+        self.avoid_rmq = self.args.avoid_rmq
+
+        Conf.load("sspl", sspl_config_path)
+        Conf.load("sspl_test", sspl_test_config_path)
+
+        # Take back up of sspl test config
+        sspl_test_backup = '/etc/sspl_tests.conf.back'
+        shutil.copyfile(sspl_test_file_path, sspl_test_backup)
+        global_config_url = Conf.get("sspl", "SYSTEM_INFORMATION>global_config_url")
+
+        # Global config path in sspl_tests.conf will be referred by sspl_tests later
+        Conf.copy("global_config", "sspl_test")
+        Conf.set("sspl", "SYSTEM_INFORMATION>global_config_url", sspl_test_config_path)
+        Conf.save("sspl")
+
+        # Enable & disable sensors based on environment
+        update_sensor_info('sspl_test')
+
+        # Get rabbitmq values from sspl.conf and update sspl_tests.conf
+        rmq_passwd = Conf.get("sspl", "RABBITMQEGRESSPROCESSOR>password")
+        Conf.set("sspl_test", "RABBITMQEGRESSPROCESSOR>password", rmq_passwd)
+        Conf.set("sspl_test", "RABBITMQINGRESSPROCESSOR>password", rmq_passwd)
+        Conf.set("sspl_test", "LOGGINGPROCESSOR>password", rmq_passwd)
+        Conf.save("sspl_test")
+
+        # TODO: Convert shell script to python
+        # from cortx.sspl.sspl_test.run_qa_test import RunQATest
+        # RunQATest(self.plan, self.avoid_rmq).run()
+        CMD = "%s/run_qa_test.sh %s %s" % (TEST_DIR, self.plan, self.avoid_rmq)
         output, error, returncode = SimpleProcess(CMD).run(realtime_output=True)
+        # Restore the original path/file & service, then throw exception
+        # if execution is failed.
+        Conf.set("sspl", "SYSTEM_INFORMATION>global_config_url", global_config_url)
+        Conf.save("sspl")
+        shutil.copyfile(sspl_test_backup, sspl_test_file_path)
+        Service('dbus').process('restart', 'sspl-ll.service')
```

Comment:
sspl restart would be needed after changing config & before starting tests also right?

Reply:
run_qa_test.sh is doing restart before starting start. But it doesn't do after the test completion. So I have added here.

Reply:
Ok, then we should do this restart too there, so in one file whole flow is captured.

Reply:
Ok i see why you are doing restart here. Reverting back config back first, after which restart needed.

Reply:
Yes, first we do config revert in line 76 and then restart in line 77.